crates/red_knot_python_semantic/src/types/statistics.rs

```diff
@@ -0,0 +1,121 @@
+use crate::types::{infer_scope_types, semantic_index, Type};
+use crate::Db;
+use ruff_db::files::File;
+use rustc_hash::FxHashMap;
+
+/// Get type-coverage statistics for a file.
+#[salsa::tracked(return_ref)]
```

Comment:
Do we expect this method to be called multiple times in the same revision and file for it to be a salsa query?

Reply:
Perhaps roughly equally likely as `check_types`, which is a query? It's very parallel to `check_types`, so seems reasonable to handle them the same way.